include/ipr/interface

```diff
@@ -667,7 +667,8 @@ namespace ipr {
 
    constexpr DeclSpecifiers operator|(DeclSpecifiers a, DeclSpecifiers b)
    {
-      return DeclSpecifiers(uint32_t(a) | uint32_t(b));
+      return static_cast<DeclSpecifiers>(
+         std::underlying_type_t<DeclSpecifiers>(a) | std::underlying_type_t<DeclSpecifiers>(b));
```

Comment:
Isn't this still C style cast? I wonder if it would be worth introducing a quick alternative to the C++23 `std::to_underlying` in IPR temporarily.

Reply:
> Isn't this still C style cast?

It is a _functional cast_ - technically not a C style cast which is written as `(T)e`.


> I wonder if it would be worth introducing a quick alternative to the C++23 std::to_underlying in IPR temporarily.

A long time ago (circa 2014), I introduced `bits::rep` and `bits::per` in the MSVC codebase to support exactly this.  In fact, I would rather introduce the conceptified version of the enum utilities I wrote there.

Reply:
I can revert the enum changes for now as that would make more sense in a separate PR.

Reply:
> I can revert the enum changes for now as that would make more sense in a separate PR.

+1.